communicator/ssh/communicator.go

```diff
@@ -344,10 +364,69 @@ func (c *comm) reconnect() (err error) {
 		c.client = ssh.NewClient(sshConn, sshChan, req)
 	}
 	c.connectToAgent()
+	c.connectTunnels(sshConn)
 
 	return
 }
 
+func (c *comm) connectTunnels(sshConn ssh.Conn) {
+	if c.client == nil {
+		return
+	}
+
+	// Start remote forwards of ports to ourselves.
+	log.Printf("[DEBUG] Tunnel Configuration: %v", c.config.Tunnels)
```

Comment:
Let's check whether there are any tunnels before adding this debug line, just to prevent clutter.